### PR TITLE
Python 3 - pathod

### PR DESCRIPTION
**Pull request by dufferzafar**
FINALLY!

### CODE REVIEW

File: test/pathod/tutils.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def get(self, spec):
         )
         with c.connect():
             resp = c.request(
-                "get:/p/%s" % urllib.parse.quote(spec).encode("string_escape")
+                "get:/p/%s" % urllib.parse.quote(spec)
```
Comment (dufferzafar) [inline test/pathod/tutils.py]: I know this is wrong, but don't know what it should be replaced with.

File: test/pathod/test_pathod.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def test_connect(self):
 class TestSSLCN(tutils.DaemonTests):
     ssl = True
     ssloptions = dict(
-        cn="foo.com"
+        cn=b"foo.com"
```
Comment (Kriechi) [inline test/pathod/test_pathod.py]: Does this mean we cannot handle UTF8 domain names?

Comment (mhils) [inline test/pathod/test_pathod.py]: The OpenSSL API expects bytes, so this change is correct by all means :smiley:
For UTF8 domains, IDNA encoding would be the way to go.

**Comment by mhils** [comment]
:tada:

Looking at the tox diff, this is fantastic. I'm super excited to actually build a precompiled Python 3.5 binary for pathod. :smiley:

File: test/pathod/tutils.py (hunk 1 above)
Comment (dufferzafar) [inline test/pathod/tutils.py]: @mhils @Kriechi @cortesi ?

**Comment by dufferzafar** [comment]
All tests pass locally. Ready for review!

File: pathod/pathoc.py [hunk 3]
```diff
@@ -241,8 +241,8 @@ def __init__(
 
     def http_connect(self, connect_to):
         self.wfile.write(
-            'CONNECT %s:%s HTTP/1.1\r\n' % tuple(connect_to) +
-            '\r\n'
+            b'CONNECT %s:%d HTTP/1.1\r\n' % (connect_to[0].encode(), connect_to[1]) +
```
Comment (mhils) [inline pathod/pathoc.py]: I think we should apply IDNA encoding for the host here.

File: test/pathod/tutils.py (hunk 1 above)
Comment (mhils) [inline test/pathod/tutils.py]: This only affects the tests, and if they all pass, this is good enough :smiley:

File: tox.ini [hunk 4]
```diff
@@ -7,7 +7,7 @@ deps =
   codecov>=2.0.5
 passenv = CI TRAVIS_BUILD_ID TRAVIS TRAVIS_BRANCH TRAVIS_JOB_NUMBER TRAVIS_PULL_REQUEST TRAVIS_JOB_ID TRAVIS_REPO_SLUG TRAVIS_COMMIT
 setenv =
-    PY3TESTS =  test/netlib test/mitmproxy/script test/pathod/test_utils.py test/pathod/test_log.py test/pathod/test_language_generators.py test/pathod/test_language_writer.py test/pathod/test_language_base.py test/pathod/test_language_http.py test/pathod/test_language_websocket.py test/pathod/test_language_http2.py test/pathod/test_pathoc.py
+    PY3TESTS =  test/netlib test/pathod/ test/mitmproxy/script
```
Comment (mhils) [inline tox.ini]: :tada: :tada: :tada:

**Comment by mhils** [comment]
FINALLY! Fixed https://github.com/mitmproxy/mitmproxy/pull/1259#discussion_r67442122, changed the commit message for https://github.com/mitmproxy/mitmproxy/pull/1259#discussion_r67119973, merged into master. All pathod tests now pass on Python 3! :smiley: :cake: :tada:

**Pull request closed without merging**